src/main/java/nextstep/subway/service/LineService.java

```diff
@@ -0,0 +1,75 @@
+package nextstep.subway.service;
+
+import nextstep.subway.domain.Line;
+import nextstep.subway.domain.LineRepository;
+import nextstep.subway.domain.Station;
+import nextstep.subway.dto.LineRequest;
+import nextstep.subway.dto.LineResponse;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional(readOnly = true)
+public class LineService {
+
+    private final LineRepository lineRepository;
+    private final StationService stationService;
+
+    public LineService(LineRepository lineRepository, StationService stationService) {
+        this.lineRepository = lineRepository;
+        this.stationService = stationService;
+    }
+
+    @Transactional
+    public LineResponse create(LineRequest request) {
+        Line line = lineRepository.save(getLine(request));
+        return LineResponse.of(line);
+    }
+
+    private Line getLine(LineRequest request) {
+        Station upStation = stationService.findById(request.getUpStationId());
+        Station downStation = stationService.findById(request.getDownStationId());
+        return new Line(request.getName(), request.getColor(), upStation, downStation, request.getDistance());
+    }
+
+    public List<LineResponse> findAllLines() {
+        List<Line> stations = lineRepository.findAll();
+
+        return stations.stream()
+                .map(station -> LineResponse.of(station))
```

Comment:
```suggestion
                .map(LineResponse::of)
```
메서드 레퍼런스를 활용할 수 있겠네요 :)